v03_pipeline/lib/reference_datasets/misc.py

```diff
@@ -78,8 +77,29 @@ def download_zip_file(url, suffix='.zip'):
     with tempfile.NamedTemporaryFile(
         suffix=suffix,
     ) as tmp_file, requests.get(url, stream=True, timeout=10) as r:
-        shutil.copyfileobj(r.raw, tmp_file)
+        tmp_file.write(r.content)
```

Comment:
hrm, this is weird.  investigating!

Reply:
Ok! I figured out what's going on here.  The `.zip` file we added here is sneakily `gzip`'ed ('Content-Encoding': 'gzip') in addition to being a `zip` archive:

```
In [84]: r = requests.get('https://www.biorxiv.org/content/biorxiv/early/2023/01/27/2022.12.16.520778/DC3/embed/media-3.zip', stream=True, timeout=10)
In [85]: len(r.raw.read(decode_content=True))
Out[85]: 332443
In [86]: r = requests.get('https://www.biorxiv.org/content/biorxiv/early/2023/01/27/2022.12.16.520778/DC3/embed/media-3.zip', stream=True, timeout=10)
In [87]: len(r.raw.read())
Out[87]: 319854
```

The `shuttle.copyfileobj` source does not decode the content, so we're left with a double compressed tmp file.  This approach is ok.  An alternative might be to just roll our own `copyfileobj` to keep the chunked approach.  Here's the existing method:  

```
def copyfileobj(fsrc, fdst, length=16*1024):
    """copy data from file-like object fsrc to file-like object fdst"""
    while 1:
        buf = fsrc.read(length)
        if not buf:
            break
        fdst.write(buf)
```

we could just add `decode_content=True` to `read` and it would work.

Reply:
Thanks for looking into this. It's probably better if we keep the chunking approach, so I added the new `copyfileobj` implementation.